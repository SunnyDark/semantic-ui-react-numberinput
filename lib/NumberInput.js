"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _semanticUiReact = require("semantic-ui-react");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var buttonStyle = {
  borderRadius: 0,
  lineHeight: 'calc(1em + 2px)',
  margin: 0
};
var inputStyle = {
  borderRadius: 0,
  textAlign: 'right'
}; // noinspection JSUnusedGlobalSymbols

var NumberInput =
/*#__PURE__*/
function (_React$Component) {
  _inherits(NumberInput, _React$Component);

  function NumberInput() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, NumberInput);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(NumberInput)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "decrementValue", function () {
      var _this$props = _this.props,
          minValue = _this$props.minValue,
          onChange = _this$props.onChange,
          stepCount = _this$props.stepCount,
          value = _this$props.value;
      var valueAsInteger = parseInt(value, 10);

      if (Number.isSafeInteger(valueAsInteger)) {
        var newValueAsInteger = valueAsInteger - stepCount;

        if (newValueAsInteger >= minValue) {
          onChange(newValueAsInteger.toString());
        }
      }
    });

    _defineProperty(_assertThisInitialized(_this), "incrementValue", function () {
      var _this$props2 = _this.props,
          maxValue = _this$props2.maxValue,
          onChange = _this$props2.onChange,
          stepCount = _this$props2.stepCount,
          value = _this$props2.value;
      var valueAsInteger = parseInt(value, 10);

      if (Number.isSafeInteger(valueAsInteger)) {
        var newValueAsInteger = valueAsInteger + stepCount;

        if (newValueAsInteger <= maxValue) {
          onChange(newValueAsInteger.toString());
        }
      }
    });

    _defineProperty(_assertThisInitialized(_this), "changeValue", function (_ref) {
      var newValue = _ref.target.value;
      var _this$props3 = _this.props,
          maxValue = _this$props3.maxValue,
          minValue = _this$props3.minValue,
          onChange = _this$props3.onChange;
      var newValueAsInteger = parseInt(newValue, 10);

      if (Number.isSafeInteger(newValueAsInteger)) {
        if (newValueAsInteger >= minValue && newValueAsInteger <= maxValue) {
          onChange(newValueAsInteger.toString());
        }
      }
    });

    return _this;
  }

  _createClass(NumberInput, [{
    key: "render",
    value: function render() {
      var _this$props4 = this.props,
          className = _this$props4.className,
          id = _this$props4.id,
          maxLength = _this$props4.maxLength,
          maxValue = _this$props4.maxValue,
          minValue = _this$props4.minValue,
          size = _this$props4.size,
          stepCount = _this$props4.stepCount,
          value = _this$props4.value;
      var valueAsInteger = parseInt(value, 10);
      var valueIsNotANumber = Number.isNaN(valueAsInteger);
      return _react["default"].createElement("div", {
        id: id,
        className: className
      }, _react["default"].createElement(_semanticUiReact.Button, {
        size: size,
        style: buttonStyle,
        icon: "minus",
        onClick: this.decrementValue,
        disabled: valueIsNotANumber || valueAsInteger - stepCount <= minValue
      }), _react["default"].createElement("div", {
        className: "ui input ".concat(size)
      }, _react["default"].createElement("input", {
        type: "text",
        style: inputStyle,
        maxLength: maxLength,
        value: value,
        onChange: this.changeValue
      })), _react["default"].createElement(_semanticUiReact.Button, {
        size: size,
        style: buttonStyle,
        icon: "plus",
        onClick: this.incrementValue,
        disabled: valueIsNotANumber || valueAsInteger + stepCount >= maxValue || (valueAsInteger + stepCount).toString().length > maxLength
      }));
    }
  }]);

  return NumberInput;
}(_react["default"].Component);

exports["default"] = NumberInput;

_defineProperty(NumberInput, "propTypes", {
  id: _propTypes["default"].string,
  // eslint-disable-next-line react/require-default-props
  value: function value(props) {
    if (!props.value) {
      return new Error('value is required');
    }

    if (!Number.isSafeInteger(parseInt(props.value, 10))) {
      return new Error('value must be a string that can be parsed to an integer');
    }

    return null;
  },
  onChange: _propTypes["default"].func.isRequired,
  className: _propTypes["default"].string,
  minValue: function minValue(props) {
    if (!Number.isSafeInteger(props.minValue)) {
      return new Error('minValue must be an integer');
    }

    if (props.minValue >= props.maxValue) {
      return new Error('maxValue must greater than minValue');
    }

    return null;
  },
  maxValue: function maxValue(props) {
    if (!Number.isSafeInteger(props.maxValue)) {
      return new Error('maxValue must be an integer');
    }

    if (props.minValue >= props.maxValue) {
      return new Error('maxValue must greater than minValue');
    }

    return null;
  },
  maxLength: function maxLength(props) {
    if (!Number.isSafeInteger(props.maxLength) || props.maxLength < 1) {
      return new Error('maxLength must be a positive integer');
    }

    return null;
  },
  size: _propTypes["default"].oneOf(['mini', 'small', 'large', 'big', 'huge', 'massive']),
  stepCount: function stepCount(props) {
    if (!Number.isSafeInteger(props.stepCount) || props.stepCount < 1) {
      return new Error('stepCount must be a positive integer');
    }

    return null;
  }
});

_defineProperty(NumberInput, "defaultProps", {
  id: null,
  className: null,
  minValue: Number.MIN_SAFE_INTEGER,
  maxValue: Number.MAX_SAFE_INTEGER,
  maxLength: 10,
  size: 'small',
  stepCount: 1
});
//# sourceMappingURL=NumberInput.js.map