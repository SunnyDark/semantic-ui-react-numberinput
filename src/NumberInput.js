// @flow

import React from 'react';
import type { Element } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';

type Props = {
  id: string,
  value: string,
  onChange: (newValue: string) => void,
  className: string,
  minValue: number,
  maxValue: number,
  maxLength: number,
  size: 'mini' | 'small' | 'large' | 'big' | 'huge' | 'massive',
  stepCount: number
};

const buttonStyle = {
  borderRadius: 0,
  lineHeight: 'calc(1em + 2px)',
  margin: 0
};

const inputStyle = {
  borderRadius: 0,
  textAlign: 'right'
};

// noinspection JSUnusedGlobalSymbols
export default class NumberInput extends React.Component<Props, {}> {
  static propTypes = {
    id: PropTypes.string,
    // eslint-disable-next-line react/require-default-props
    value(props: Props): ?Error {
      if (!props.value) {
        return new Error('value is required');
      }
      if (!Number.isSafeInteger(parseInt(props.value, 10))) {
        return new Error('value must be a string that can be parsed to an integer');
      }
      return null;
    },
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string,
    minValue(props: Props): ?Error {
      if (!Number.isSafeInteger(props.minValue)) {
        return new Error('minValue must be an integer');
      }
      if (props.minValue >= props.maxValue) {
        return new Error('maxValue must greater than minValue');
      }
      return null;
    },
    maxValue(props: Props): ?Error {
      if (!Number.isSafeInteger(props.maxValue)) {
        return new Error('maxValue must be an integer');
      }
      if (props.minValue >= props.maxValue) {
        return new Error('maxValue must greater than minValue');
      }
      return null;
    },
    maxLength(props: Props): ?Error {
      if (!Number.isSafeInteger(props.maxLength) || props.maxLength < 1) {
        return new Error('maxLength must be a positive integer');
      }
      return null;
    },
    size: PropTypes.oneOf(['mini', 'small', 'large', 'big', 'huge', 'massive']),
    stepCount(props: Props): ?Error {
      if (!Number.isSafeInteger(props.stepCount) || props.stepCount < 1) {
        return new Error('stepCount must be a positive integer');
      }
      return null;
    }
  };

  static defaultProps = {
    id: null,
    className: null,
    minValue: Number.MIN_SAFE_INTEGER,
    maxValue: Number.MAX_SAFE_INTEGER,
    maxLength: 10,
    size: 'small',
    stepCount: 1
  };

  decrementValue = () => {
    const { minValue, onChange, stepCount, value } = this.props;
    const valueAsInteger = parseInt(value, 10);

    if (Number.isSafeInteger(valueAsInteger)) {
      const newValueAsInteger = valueAsInteger - stepCount;
      if (newValueAsInteger >= minValue) {
        onChange(newValueAsInteger.toString());
      }
    }
  };

  incrementValue = () => {
    const { maxValue, onChange, stepCount, value } = this.props;
    const valueAsInteger = parseInt(value, 10);

    if (Number.isSafeInteger(valueAsInteger)) {
      const newValueAsInteger = valueAsInteger + stepCount;
      if (newValueAsInteger <= maxValue) {
        onChange(newValueAsInteger.toString());
      }
    }
  };

  changeValue = ({ target: { value: newValue } }: SyntheticInputEvent<HTMLInputElement>) => {
    const { maxValue, minValue, onChange } = this.props;
    const newValueAsInteger = parseInt(newValue, 10);

    if (Number.isSafeInteger(newValueAsInteger)) {
      if (newValueAsInteger >= minValue && newValueAsInteger <= maxValue) {
        onChange(newValueAsInteger.toString());
      }
    }
  };

  render(): Element<*> {
    const { className, id, maxLength, maxValue, minValue, size, stepCount, value } = this.props;
    const valueAsInteger = parseInt(value, 10);
    const valueIsNotANumber = Number.isNaN(valueAsInteger);

    return (
      <div id={id} className={className}>
        <Button
          size={size}
          style={buttonStyle}
          icon="minus"
          onClick={this.decrementValue}
          disabled={valueIsNotANumber || valueAsInteger - stepCount <= minValue}
        />
        <div className={`ui input ${size}`}>
          <input
            type="text"
            style={inputStyle}
            maxLength={maxLength}
            value={value}
            onChange={this.changeValue}
          />
        </div>
        <Button
          size={size}
          style={buttonStyle}
          icon="plus"
          onClick={this.incrementValue}
          disabled={
            valueIsNotANumber ||
            valueAsInteger + stepCount >= maxValue ||
            (valueAsInteger + stepCount).toString().length > maxLength
          }
        />
      </div>
    );
  }
}
