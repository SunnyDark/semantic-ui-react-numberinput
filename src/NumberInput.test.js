import React from 'react';
import { mount, shallow as renderShallow } from 'enzyme';
import { Button } from 'semantic-ui-react';
import NumberInput from './NumberInput';

let onChangeMock;
let consoleErrorSpy;
const NON_INTEGER_VALUE = 1.1;

beforeEach(() => {
  onChangeMock = jest.fn();
  consoleErrorSpy = jest.spyOn(global.console, 'error');
});

test('NumberInput should render correctly', () => {
  const numberInput = renderShallow(<NumberInput value="1" onChange={onChangeMock} />);
  expect(numberInput).toMatchSnapshot();
});

test('NumberInput should render correctly when buttonPlacement is right', () => {
  const numberInput = renderShallow(
    <NumberInput buttonPlacement="right" value="1" onChange={onChangeMock} />
  );
  expect(numberInput).toMatchSnapshot();
});

test('NumberInput sets default props correctly', () => {
  const numberInput = mount(<NumberInput value="1" onChange={onChangeMock} />);

  expect(numberInput.props().value).toBe('1');
  expect(numberInput.props().onChange).toBe(onChangeMock);
  expect(numberInput.props().id).toBe(null);
  expect(numberInput.props().buttonPlacement).toBe('leftAndRight');
  expect(numberInput.props().className).toBe(null);
  expect(numberInput.props().minValue).toBe(-999999999);
  expect(numberInput.props().maxValue).toBe(9999999999);
  expect(numberInput.props().maxLength).toBe(10);
  expect(numberInput.props().precision).toBe(2);
  expect(numberInput.props().size).toBe('small');
  expect(numberInput.props().stepAmount).toBe(1);
  expect(numberInput.props().valueType).toBe('integer');
});

test('pressing NumberInput´s decrement button decrements value (valueType is integer)', () => {
  const numberInput = mount(<NumberInput value="1" onChange={onChangeMock} />);
  const decrementButton = numberInput.find('button').first();

  decrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('0');
});

test('pressing NumberInput´s decrement button decrements value (valueType is decimal)', () => {
  const numberInput = mount(
    <NumberInput value="2.0" valueType="decimal" precision={2} stepAmount={0.25} onChange={onChangeMock} />
  );
  const decrementButton = numberInput.find('button').first();

  decrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('1.75');
});

test('pressing NumberInput´s increment button increments value (valueType is integer)', () => {
  const numberInput = mount(<NumberInput value="1" onChange={onChangeMock} />);
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('2');
});

test('pressing NumberInput´s increment button increments value (valueType is decimal)', () => {
  const numberInput = mount(<NumberInput value="0.12" valueType="decimal" precision={1} stepAmount={0.1} onChange={onChangeMock} />);
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('0.2');
});

test('NumberInput´s increment button with negative value', () => {
  const numberInput = mount(<NumberInput value="-2" onChange={onChangeMock} />);
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('-1');
});

test('pressing NumberInput´s decrement button decrements value by given stepAmount', () => {
  const numberInput = mount(<NumberInput value="3" stepAmount={2} onChange={onChangeMock} />);
  const decrementButton = numberInput.find('button').first();

  decrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('1');
});

test('pressing NumberInput´s increment button increments value by given stepAmount', () => {
  const numberInput = mount(<NumberInput value="1" stepAmount={2} onChange={onChangeMock} />);
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('3');
});

test('NumberInput´s decrement button with negative value', () => {
  const numberInput = mount(<NumberInput value="-2" onChange={onChangeMock} />);
  const decrementButton = numberInput.find('button').first();

  decrementButton.simulate('click');

  expect(onChangeMock).toHaveBeenCalledWith('-3');
});

test('pressing NumberInput`s decrement button does not decrement value beyond minimum safe integer value', () => {
  const numberInput = mount(
    <NumberInput value={Number.MIN_SAFE_INTEGER.toString()} onChange={onChangeMock} />
  );
  const decrementButton = numberInput.find('button').first();

  decrementButton.simulate('click');

  expect(onChangeMock).not.toHaveBeenCalled();
});

test('pressing NumberInput´s increment button does not increment value beyond maximum safe integer value', () => {
  const numberInput = mount(
    <NumberInput value={Number.MAX_SAFE_INTEGER.toString()} onChange={onChangeMock} />
  );
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).not.toHaveBeenCalled();
});

test('pressing NumberInput´s decrement button does not decrement value if current value is minValue', () => {
  const numberInput = mount(<NumberInput value="5" minValue={5} onChange={onChangeMock} />);
  const decrementButton = numberInput.find('button').first();

  decrementButton.simulate('click');

  expect(onChangeMock).not.toHaveBeenCalled();
  expect(decrementButton.hasClass('disabled')).toBe(true);
});

test('pressing NumberInput´s decrement button does not decrement value beyond minValue', () => {
  const numberInput = mount(<NumberInput value="5" minValue={4} stepAmount={2} onChange={onChangeMock} />);
  const decrementButton = numberInput.find('button').first();

  decrementButton.simulate('click');

  expect(onChangeMock).not.toHaveBeenCalled();
  expect(decrementButton.hasClass('disabled')).toBe(true);
});

test('pressing NumberInput´s increment button does not increment value if current value is maxValue', () => {
  const numberInput = mount(<NumberInput value="5" maxValue={5} onChange={onChangeMock} />);
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).not.toHaveBeenCalled();
  expect(incrementButton.hasClass('disabled')).toBe(true);
});

test('pressing NumberInput´s increment button does not increment value beyond given maxValue', () => {
  const numberInput = mount(<NumberInput value="4" maxValue={5} stepAmount={2} onChange={onChangeMock} />);
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).not.toHaveBeenCalled();
  expect(incrementButton.hasClass('disabled')).toBe(true);
});

test('changing NumberInput´s input value (valueType is integer)', () => {
  const numberInput = mount(<NumberInput value="1" onChange={onChangeMock} />);
  const input = numberInput.find('input');

  input.simulate('change', { target: { value: '3' } });

  expect(onChangeMock).toHaveBeenCalledWith('3');
});

test('changing NumberInput´s input value (valueType is decimal)', () => {
  const numberInput = mount(<NumberInput value="1" valueType="decimal" precision={2} onChange={onChangeMock} />);
  const input = numberInput.find('input');

  input.simulate('change', { target: { value: '3.245' } });

  expect(onChangeMock).toHaveBeenCalledWith('3.25');
});

test('trying to change NumberInput´s input value beyond minValue', () => {
  const numberInput = mount(<NumberInput value="1" minValue={0} onChange={onChangeMock} />);
  const input = numberInput.find('input');

  input.simulate('change', { target: { value: '-1' } });

  expect(onChangeMock).not.toHaveBeenCalled();
});

test('trying to change NumberInput´s input value beyond maxValue', () => {
  const numberInput = mount(<NumberInput value="1" maxValue={10} onChange={onChangeMock} />);
  const input = numberInput.find('input');

  input.simulate('change', { target: { value: '11' } });

  expect(onChangeMock).not.toHaveBeenCalled();
});

test('trying to change NumberInput´s input value to non-numeric value', () => {
  const numberInput = mount(<NumberInput value="1" onChange={onChangeMock} />);
  const input = numberInput.find('input');

  input.simulate('change', { target: { value: 'x' } });

  expect(onChangeMock).not.toHaveBeenCalled();
});

test('className prop sets className for outer div', () => {
  const numberInput = mount(<NumberInput className="test" value="1" onChange={onChangeMock} />);
  const outerDiv = numberInput.find('div').first();

  expect(outerDiv.hasClass('test')).toBe(true);
});

test('id prop sets id for outer div', () => {
  const numberInput = mount(<NumberInput id="numberInput1" value="1" onChange={onChangeMock} />);
  expect(numberInput.exists('div#numberInput1')).toBe(true);
});

test('NumberInput should work correctly when trying to change input value beyond maxLength', () => {
  const numberInput = mount(<NumberInput value="99" maxLength={2} onChange={onChangeMock} />);
  const input = numberInput.find('input');

  input.simulate('focus');
  input.simulate('keypress', { key: '9' });

  expect(input.props().value).toBe('99');
  expect(onChangeMock).not.toHaveBeenCalled();
});

test('NumberInput should work correctly when trying to increment value beyond maxLength', () => {
  const numberInput = mount(<NumberInput value="99" maxLength={2} onChange={onChangeMock} />);
  const incrementButton = numberInput.find('button').last();

  incrementButton.simulate('click');

  expect(onChangeMock).not.toHaveBeenCalled();
  expect(incrementButton.hasClass('disabled')).toBe(true);
});

test('size prop is correctly propagated to sub components', () => {
  const numberInput = mount(<NumberInput value="99" size="mini" onChange={onChangeMock} />);
  const decrementButton = numberInput.find(Button).first();
  const incrementButton = numberInput.find(Button).last();
  const inputDiv = numberInput.find('div > div');

  expect(decrementButton.props().size).toBe('mini');
  expect(incrementButton.props().size).toBe('mini');
  expect(inputDiv.hasClass('mini')).toBe(true);
});

test('value prop is valid', () => {
  renderShallow(<NumberInput value="" onChange={onChangeMock} />);
  renderShallow(<NumberInput value="x" onChange={onChangeMock} />);

  expect(consoleErrorSpy).toHaveBeenCalledWith(expect.stringContaining('value is required'));
  expect(consoleErrorSpy).toHaveBeenCalledWith(
    expect.stringContaining('value must be a string that can be parsed to an integer')
  );
});

test('maxValue prop is valid', () => {
  renderShallow(<NumberInput value="1" maxValue={NON_INTEGER_VALUE} onChange={onChangeMock} />);
  renderShallow(<NumberInput value="1" minValue={0} maxValue={100} maxLength={2} onChange={onChangeMock} />);

  expect(consoleErrorSpy).toHaveBeenCalledWith(expect.stringContaining('maxValue must be an integer'));
  expect(consoleErrorSpy).toHaveBeenCalledWith(expect.stringContaining('maxValue does not fit in maxLength'));
});

test('minValue prop is valid', () => {
  renderShallow(<NumberInput value="1" minValue={NON_INTEGER_VALUE} onChange={onChangeMock} />);
  renderShallow(<NumberInput value="1" minValue={-99} maxValue={99} maxLength={2} onChange={onChangeMock} />);

  expect(consoleErrorSpy).toHaveBeenCalledWith(expect.stringContaining('minValue must be an integer'));
  expect(consoleErrorSpy).toHaveBeenCalledWith(expect.stringContaining('minValue does not fit in maxLength'));
});

test('minValue cannot be greater than maxValue', () => {
  renderShallow(<NumberInput value="1" minValue={2} maxValue={1} onChange={onChangeMock} />);

  expect(consoleErrorSpy).toHaveBeenCalledWith(
    expect.stringContaining('maxValue must greater than or equal to minValue')
  );
});

test('maxLength prop must be a positive integer', () => {
  renderShallow(<NumberInput value="1" maxLength={0} onChange={onChangeMock} />);

  expect(consoleErrorSpy).toHaveBeenCalledWith(
    expect.stringContaining('maxLength must be a positive integer')
  );
});

test('minValue and maxValue must fit in maxLength', () => {
  renderShallow(<NumberInput value="1" maxLength={1} onChange={onChangeMock} />);

  expect(consoleErrorSpy).toHaveBeenCalledWith(expect.stringContaining('maxValue does not fit in maxLength'));
  expect(consoleErrorSpy).toHaveBeenCalledWith(expect.stringContaining('minValue does not fit in maxLength'));
});

test('stepAmount prop is valid', () => {
  renderShallow(<NumberInput value="1" stepAmount={-2} onChange={onChangeMock} />);

  expect(consoleErrorSpy).toHaveBeenCalledWith(
    expect.stringContaining('stepAmount must be a positive integer')
  );
});
