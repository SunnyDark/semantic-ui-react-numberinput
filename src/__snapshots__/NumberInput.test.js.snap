// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput should render correctly 1`] = `ShallowWrapper {}`;

exports[`NumberInput should render correctly when buttonPlacement is right 1`] = `ShallowWrapper {}`;
